src/bindings/c/include/c_api/ie_c_api.h

```diff
@@ -56,6 +60,16 @@
 #    define INFERENCE_ENGINE_C_API_CALLBACK
 #endif
 
+#if !defined(IN_OV_COMPONENT) && !defined(C_API_LEGACY_HEADER_INCLUDED)
```

Comment:
It is wired, but please move this to line38 like https://github.com/openvinotoolkit/openvino/pull/18370